onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -372,6 +414,95 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
   }
 }
 
+
+/*
+ * Register execution provider with options.
+ *
+ * (note: currently only cuda EP supports this feature and rest of EPs use default options) 
+ */
+void RegisterExecutionProvidersWithOptions(InferenceSession* sess, const std::vector<std::string>& provider_types, ProviderOptionsMap& provider_options_map) {
+  for (const std::string& type : provider_types) {
+    if (type == kCpuExecutionProvider) {
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_CPU(sess->GetSessionOptions().enable_cpu_mem_arena));
+    } else if (type == kTensorrtExecutionProvider) {
+#ifdef USE_TENSORRT
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_Tensorrt(0));
+#endif
+    } else if (type == kMIGraphXExecutionProvider) {
+#ifdef USE_MIGRAPHX
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_MIGraphX(0));
+#endif
+    } else if (type == kCudaExecutionProvider) {
+#ifdef USE_CUDA
+      onnxruntime::CudaProviderOptions cuda_provider_options;
```

Comment:
Where is CudaProviderOptions defined?
cuda_provider_options.h seems empty?

Reply:
i assume there are default values for the cuda options?

Reply:
Yes, there are default values for the cuda provider options defined in a file.
I forgot to update the content of that file.
Please see onnxruntime/core/providers/cuda/cuda_provider_options.h